docs/user-guide/advanced_topics/timeouts.md

```diff
@@ -0,0 +1,42 @@
+
+At the federate level a --granttimeout option is available. This depends on ASIO being compiled in and works similar to the real time mode of federates. At present it is primarily for diagnostics as it cannot generate an error, though this may change at some point in the future. It is disabled by default since some federates could legitimately take a long time to execute so having a timeout on others doesn't make sense. So any grant timeout needs to be handled based on the user knowledge of how the different federates operate. This operation is subject to change but at present the operation is handled in 4 stages.
+
+- 1X timeout print warning message
+- 3X timeout request a resend of timing message from blocking dependencies (in case of partial communications failure)
+- 6X timeout print timing diagnostic information and send message to parent for additional debugging information
+- 10X print additional warning -- this is when additional corrective action may be taken but that is a work in progress
+
+It is expected this process will change as more experience with it is gained.
+
+At the broker level a `--maxcosimduration` option is available. By default it is disabled. If specified on a broker or core the co-simulation will terminate if running for more than the specified total time. This can be useful if running a large number of co-simulations in an automated fashion and the expected time is known. This can prevent a situation where a single co-simulation hangs indefinitely blocking others and doing no useful work.
+
+## Other timeouts
+
+There are two other timeouts that can be specified. The first is a query timeout. (`--querytimeout=X`) the default is 15 seconds. This applies to queries which are blocking calls, and if a federate is shutting down or fails as a query is in flight it is possible (not likely) that it could result in a non response and the query never completing. In that rare situation the query timeout will trigger and result in an error on the query. In very large co-simulations it is possible a complex query could exceed the timeout without there being a problem so it may be necessary to lengthen it.
```

Comment:
```suggestion
There are two other timeouts that can be specified. The first is a query timeout (`--querytimeout=X`). The default is 15 seconds. This applies to queries which are blocking calls, and if a federate is shutting down or fails as a query is in flight it is possible (not likely) that it could result in a non-response and the query never completing. In that rare situation the query timeout will trigger and result in an error on the query. In very large co-simulations it is possible a complex query could exceed the timeout without there being a problem so it may be necessary to lengthen it.
```